app/lib/differential_expression_service.rb

```diff
@@ -243,7 +244,9 @@ def self.results_exist?(study, annotation)
   def self.study_eligible?(study, skip_existing: false)
     begin
       validate_study(study)
-      find_eligible_annotations(study, skip_existing:).any? && study.has_raw_counts_matrices?
+      find_eligible_annotations(study, skip_existing:).any? &&
+        study.has_raw_counts_matrices? &&
+        !study_has_author_de?(study)
```

Comment:
When I see multi-clause conditionals broken out onto separate lines in Python or JavaScript, each clause vertically aligns with the others.  Here, the clauses after the first aren't aligned, which caused me to double-take.

Non-blocking, for consideration:

```suggestion
      find_eligible_annotations(study, skip_existing:).any? &&
      study.has_raw_counts_matrices? &&
      !study_has_author_de?(study)
```

(I haven't tested that locally.)

Reply:
I take your point about readability.  That being said, the above formatting is considered conventional in Ruby: https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/MultilineOperationIndentation

My feeling would be to for us to retain idiomatic/conventional usage for each language in our stack.  It will make us all better at reading that code natively, which will pay dividends as we move forward, both in this project and others.

Another example that I know we've debated that seems very relevant: `unless` vs. `if !expression`...

Reply:
Reasonable!